File: app/src/main/java/starbright/com/projectegg/features/ingredients/IngredientsContract.java
package starbright.com.projectegg.features.ingredients;

import java.util.ArrayList;
import java.util.List;

import starbright.com.projectegg.data.local.model.Ingredient;
import starbright.com.projectegg.features.base.BasePresenter;
import starbright.com.projectegg.features.base.BaseView;

class IngredientsContract {

    interface View extends BaseView<IngredientsPresenter> {
        void setupEtSearchIngredients();
        void setupRvIngredientSuggestion();
        void setupMaterialProgressDialog();
        void setupBottomSheetDialogFragment();

        void clearEtSearchQuery();
        void openCamera();

        void showSearchSuggestion(List<Ingredient> ingredients);
        void hideSearchSuggestion();

        void showActionCamera();
        void showActionClear();
        void hideActionButton();

        void showSuggestionProgressBar();
        void hideSuggestionProgressBar();

        void showItemEmptyToast();
        void showItemMaxToast();
        void showDuplicateItemToast();
        void showPredictionEmptyToast();

        void showItemAddedToast();

        void showSuccessPutIngredientToast(String ingredientName);

        void showMaterialProgressDialog();
        void hideMaterialProgressDialog();

        void showBottomSheetDialog();

        void setCartItem(List<Ingredient> cart);

        void updateIngredientCount(int count);

        void hideSoftkeyboard();
    }

    interface Presenter extends BasePresenter {
        void handleActionButtonClicked(String query);
        void handleCartTvClicked();

        void handleSuggestionTextChanged(String query);
        void handleSuggestionItemClicked(Ingredient ingredient);
        void handleCameraResult(String filePath);

        void searchIngredient(String query);

        void setCart(List<Ingredient> cart);
        ArrayList<Ingredient> getCart();
    }
}

File: app/src/main/java/starbright/com/projectegg/features/ingredients/IngredientsContract.kt
/**
 * Created by Andreas on 9/9/2018.
 */

package starbright.com.projectegg.features.ingredients

import starbright.com.projectegg.data.model.Ingredient
import starbright.com.projectegg.features.base.BasePresenterContract
import starbright.com.projectegg.features.base.BaseViewContract

class IngredientsContract {

    interface View : BaseViewContract {
        fun setupView()

        fun clearEtSearchQuery()
        fun openCamera()

        fun updateSuggestion(ingredients: List<Ingredient>)
        fun hideSearchSuggestion()

        fun showActionCamera()
        fun showActionClear()
        fun hideActionButton()

        fun showSuggestionProgressBar()
        fun hideSuggestionProgressBar()

        fun showItemEmptyToast()
        fun showItemMaxToast()
        fun showDuplicateItemToast()
        fun showPredictionEmptyToast()

        fun showItemAddedToast()

        fun showSuccessPutIngredientToast(ingredientName: String)

        fun showMaterialProgressDialog()
        fun hideMaterialProgressDialog()

        fun showBottomSheetDialog(cart: MutableList<Ingredient>)

        fun updateIngredientCount(count: Int)

        fun hideSoftKeyboard()

        fun navigateToRecipeList(cart: List<Ingredient>)
    }

    interface Presenter : BasePresenterContract {
        fun handleActionButtonClicked(query: String)
        fun handleCartTvClicked()

        fun handleSuggestionTextChanged(query: String)
        fun handleSuggestionItemClicked(ingredient: Ingredient)
        fun handleCameraResult(filePath: String)

        fun handleSubmitButtonClicked()
        fun handleItemRemovedFromCart()

        fun searchIngredient(query: String)
    }
}